File: api/services/translation_service.py
import os
import aiohttp
from models import LANGUAGE_MAP

async def translate_text(text: str, source_language: str = "it", target_language: str = "en") -> str:
    """Traduce il testo da una lingua all'altra utilizzando Azure Translator"""
    translator_key = os.getenv("TRANSLATOR_KEY")
    translator_endpoint = os.getenv("TRANSLATOR_ENDPOINT")
    
    if not translator_key or not translator_endpoint:
        raise Exception("Chiavi Azure Translator non configurate")

    headers = {
        "Ocp-Apim-Subscription-Key": translator_key,
        "Ocp-Apim-Subscription-Region": "italynorth",
        "Content-Type": "application/json",
    }

    body = [{"text": text}]
    params = {
        "api-version": "3.0", 
        "from": LANGUAGE_MAP.get(source_language, "it-IT").split("-")[0],
        "to": LANGUAGE_MAP.get(target_language, "en-US").split("-")[0]
    }

    async with aiohttp.ClientSession() as session:
        async with session.post(
            f"{translator_endpoint}/translate",
            headers=headers,
            json=body,
            params=params
        ) as response:
            result = await response.json()
            return result[0]["translations"][0]["text"]

File: AudioServer/Services/TranslationService.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Extensions.Configuration;
using Newtonsoft.Json;

public class TranslationService
{
    private readonly HttpClient _httpClient;
    private readonly string _translatorKey;
    private readonly string _translatorEndpoint;
    private readonly string _translatorRegion;

    public TranslationService(IConfiguration configuration)
    {
        _httpClient = new HttpClient();
        _translatorKey = configuration["Azure:TranslatorKey"];
        _translatorEndpoint = configuration["Azure:TranslatorEndpoint"];
        _translatorRegion = configuration["Azure:TranslatorRegion"];
    }

    public async Task<string> TranslateTextAsync(string text, string sourceLanguage, string targetLanguage)
    {
        if (string.IsNullOrEmpty(text))
            return text;

        if (sourceLanguage == targetLanguage)
            return text;

        string route = $"/translate?api-version=3.0&from={sourceLanguage}&to={targetLanguage}";

        var requestBody = new[] { new { Text = text } };
        var requestBodyJson = JsonConvert.SerializeObject(requestBody);

        using (var request = new HttpRequestMessage())
        {
            request.Method = HttpMethod.Post;
            request.RequestUri = new Uri(_translatorEndpoint + route);
            request.Content = new StringContent(requestBodyJson, Encoding.UTF8, "application/json");
            request.Headers.Add("Ocp-Apim-Subscription-Key", _translatorKey);
            request.Headers.Add("Ocp-Apim-Subscription-Region", _translatorRegion);

            var response = await _httpClient.SendAsync(request);
            response.EnsureSuccessStatusCode();

            var responseBody = await response.Content.ReadAsStringAsync();
            var result = JsonConvert.DeserializeObject<TranslationResult[]>(responseBody);

            if (result?.Length > 0 && result[0].Translations.Length > 0)
            {
                return result[0].Translations[0].Text;
            }

            return text; // Fallback al testo originale
        }
    }

    private class TranslationResult
    {
        public Translation[] Translations { get; set; }
    }

    private class Translation
    {
        public string Text { get; set; }
        public string To { get; set; }
    }
}